components/create_post/create_post.jsx

```diff
@@ -675,17 +662,58 @@ export default class CreatePost extends React.Component {
         }
     }
 
+    comboKeyIsPressed = (e) => {
+        const ctrlOrMetaKeyPressed = e.ctrlKey || e.metaKey;
+        return ctrlOrMetaKeyPressed || e.shiftKey || e.altKey;
+    }
+
+    comboKeyToSendMessage = (e) => {
+        const ctrlOrMetaKeyPressed = e.ctrlKey || e.metaKey;
+        const comboKeyIsPressed = this.comboKeyIsPressed(e);
+        return (!this.props.ctrlSend && !comboKeyIsPressed) || (this.props.ctrlSend && ctrlOrMetaKeyPressed);
+    }
+
+    comboKeyToEnterNewLine = (e) => {
+        return this.comboKeyIsPressed(e);
+    }
+
+    addNewLine = (e) => {
+        e.preventDefault();
+        const end = e.target.selectionEnd;
+        const start = e.target.selectionStart;
+        const message = this.state.message.substring(0, start) + '\n' + this.state.message.substring(end);
+        this.setState({
+            message,
+        }, () => {
+            if (this.refs.textbox) {
+                this.refs.textbox.setInputCursorPosition(start + 1);
+            }
+        });
+        const draft = {
+            ...this.props.draft,
+            message,
+        };
+
+        this.props.actions.setDraft(StoragePrefixes.DRAFT + this.props.currentChannel.id, draft);
+    }
+
     handleKeyDown = (e) => {
         const ctrlOrMetaKeyPressed = e.ctrlKey || e.metaKey;
         const messageIsEmpty = this.state.message.length === 0;
         const draftMessageIsEmpty = this.props.draft.message.length === 0;
-        const ctrlEnterKeyCombo = this.props.ctrlSend && Utils.isKeyPressed(e, KeyCodes.ENTER) && ctrlOrMetaKeyPressed;
-        const upKeyOnly = !ctrlOrMetaKeyPressed && !e.altKey && !e.shiftKey && Utils.isKeyPressed(e, KeyCodes.UP);
+        const upKeyOnly = !this.comboKeyIsPressed(e) && Utils.isKeyPressed(e, KeyCodes.UP);
         const shiftUpKeyCombo = !ctrlOrMetaKeyPressed && !e.altKey && e.shiftKey && Utils.isKeyPressed(e, KeyCodes.UP);
         const ctrlKeyCombo = ctrlOrMetaKeyPressed && !e.altKey && !e.shiftKey;
 
-        if (ctrlEnterKeyCombo) {
-            this.postMsgKeyPress(e);
+        if (!UserAgent.isMobile() && Utils.isKeyPressed(e, KeyCodes.ENTER)) {
+            if (this.comboKeyToSendMessage(e)) {
+                e.preventDefault();
+                ReactDOM.findDOMNode(this.refs.textbox).blur();
```

Comment:
What's the reason for the blur? Will this de-focus the textbox when I send a message?

Reply:
I guess that was the behavior added previously but it does not work and it would have been odd if it worked. Gonna remove that line and check with a PM if we need to get it working and a file a ticket if needed.